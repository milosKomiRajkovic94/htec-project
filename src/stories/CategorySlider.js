/* eslint-disable prefer-const */
import React, { useState } from "react";

import PropTypes from "prop-types";

import { capitalizeFirstLetter } from "../utils/capitalize-first-letter";

import "./categorySlider.css";

import "simple-line-icons/css/simple-line-icons.css";

import { Article } from "./Article";

export const CategorySlider = ({
  onCategoryClick,
  onMoreClick,
  dataArray,
  categoryTitle,
}) => {
  const [isMenuExpanded, setIsMenuExpanded] = useState(true);
  let [currentIndex, setCurrentIndex] = useState(0);

  function onMoveLeft() {
    currentIndex -= 1;
    setCurrentIndex(currentIndex);
  }

  function onMoveRight() {
    currentIndex += 1;
    setCurrentIndex(currentIndex);
  }

  return (
    <div className="categorySliderContainer">
      <div className="categorySliderTitle">
        <span className="categorySliderTitleSpan cursorPointer">
          <li />
          <span
            onClick={() => onCategoryClick(categoryTitle.toLowerCase())}
            onKeyPress={() => onCategoryClick()}
            role="button"
            tabIndex={0}
          >
            {capitalizeFirstLetter(categoryTitle.toLowerCase())}
          </span>
        </span>
        <i
          className={
            isMenuExpanded
              ? "icon-arrow-down cursorPointer expandIcon noselect"
              : "icon-arrow-up cursorPointer expandIcon noselect"
          }
          onClick={() => setIsMenuExpanded(!isMenuExpanded)}
          onKeyPress={() => setIsMenuExpanded(!isMenuExpanded)}
          tabIndex={0}
          role="button"
        />
      </div>
      <div
        className={
          isMenuExpanded ? "categorySlidesFuncContainer" : "displayNone"
        }
      >
        <span
          className={
            currentIndex !== 0
              ? "categorySliderMoverLeft cursorPointer noselect"
              : "categorySliderMoveLeft cursorPointer visibilityHidden noselect"
          }
          onKeyPress={() => onMoveLeft()}
          onClick={() => onMoveLeft()}
          tabIndex={0}
          role="button"
          data-testid="moveLeftButton"
        >
          {"<"}
        </span>
        <div className="categorySlidesContainer">
          {dataArray.map((news, index) => {
            return (
              <span
                className={
                  currentIndex === index ||
                  currentIndex + 1 === index ||
                  currentIndex + 2 === index
                    ? "categoryArticle animated fadeIn"
                    : "categoryArticle displayNone"
                }
                key={categoryTitle + news.title + news.description}
              >
                <Article
                  singleArticleData={news}
                  onMore={() => onMoreClick(news)}
                />
              </span>
            );
          })}
        </div>
        <span
          className={
            currentIndex !== dataArray.length - 1
              ? "categorySliderMoverRight cursorPointer noselect"
              : "categorySliderMoverRight cursorPointer visibilityHidden noselect"
          }
          onKeyPress={() => onMoveRight()}
          onClick={() => onMoveRight()}
          tabIndex={0}
          role="button"
          data-testid="moveRightButton"
        >
          {">"}
        </span>
      </div>
    </div>
  );
};

CategorySlider.propTypes = {
  onCategoryClick: PropTypes.func,
  onMoreClick: PropTypes.func,
  dataArray: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      urlToImage: PropTypes.string,
      description: PropTypes.string,
    })
  ),
  categoryTitle: PropTypes.string,
};

CategorySlider.defaultProps = {
  onCategoryClick: () => {},
  onMoreClick: () => {},
  dataArray: [],
  categoryTitle: "",
};
