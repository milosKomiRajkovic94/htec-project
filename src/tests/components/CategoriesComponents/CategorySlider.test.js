import React from "react";

import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import CategorySlider from "../../../components/CategoriesComponents/CategorySlider";

afterEach(cleanup);

it("renders without crashing withouth props", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CategorySlider />, div);
});

it("renders navigation buttons correctly", () => {
  const { getByTestId } = render(<CategorySlider />);
  expect(getByTestId("moveLeftButton")).toHaveTextContent("<");
  expect(getByTestId("moveRightButton")).toHaveTextContent(">");
});

it("matches snapshot", () => {
  const tree = renderer.create(<CategorySlider />).toJSON();
  expect(tree).toMatchSnapshot();
});
