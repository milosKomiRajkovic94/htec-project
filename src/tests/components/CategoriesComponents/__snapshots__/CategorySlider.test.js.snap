// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="categorySliderContainer"
>
  <div
    className="categorySliderTitle"
  >
    <span
      className="categorySliderTitleSpan cursorPointer"
    >
      <li />
      <span
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        
      </span>
    </span>
    <i
      className="icon-arrow-down cursorPointer expandIcon noselect"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
  <div
    className="categorySlidesFuncContainer"
  >
    <span
      className="categorySliderMoveLeft cursorPointer visibilityHidden noselect"
      data-testid="moveLeftButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      &lt;
    </span>
    <div
      className="categorySlidesContainer"
    />
    <span
      className="categorySliderMoverRight cursorPointer noselect"
      data-testid="moveRightButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      &gt;
    </span>
  </div>
</div>
`;
